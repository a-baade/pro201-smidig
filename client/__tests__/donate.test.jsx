import React, { createElement } from "react";
import ReactDOM from "react-dom";
import { MemoryRouter } from "react-router-dom";
import Donate from "../pages/donate";

describe("contact component", () => {
  it("should test div element", async function () {
    const divElement = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <Donate />
      </MemoryRouter>,
      divElement
    );
    expect(divElement.innerHTML).toMatchSnapshot();
  });
});
