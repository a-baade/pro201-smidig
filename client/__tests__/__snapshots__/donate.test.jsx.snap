// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contact component should test div element 1`] = `"<div class=\\"loading-wrapper\\"><div class=\\"loading\\"><div></div><div></div><div></div><div></div><div></div><div></div><div></div><div></div></div></div>"`;

exports[`contact component should test donate module 1`] = `"<div class=\\"donate-page page-margin custom-shadow\\"><a href=\\"/charities\\"><div class=\\"back-button custom-shadow-button\\"><img src=\\"[object Object]\\"></div></a><div class=\\"donate-page-top\\"><img></div><div class=\\"donate-page-content\\"><img class=\\"donate-logo custom-shadow\\"><h2>Donate to Charity 1</h2><p>Select amount (minimum $ 10.00)</p><div class=\\"donate-form-amount\\"><button class=\\"custom-shadow-button\\">$ 100.00</button><button class=\\"custom-shadow-button\\">$ 50.00</button><button class=\\"custom-shadow-button\\">$ 25.00</button><input type=\\"text\\" placeholder=\\"Other amount\\"></div><div class=\\"donate-text-div\\"><p>How often would you like to donate</p><div class=\\"information-icon\\">?<div class=\\"information-box\\">Choose how often you would like to donate selected amount to this charity. You can change amount and frequency at any time.</div></div></div><div class=\\"donate-form-frequency\\"><div><input type=\\"radio\\" id=\\"monthly\\" name=\\"donation-frequency\\" value=\\"monthly\\"><label for=\\"monthly\\">Monthly</label></div><div><input type=\\"radio\\" id=\\"quarterly\\" name=\\"donation-frequency\\" value=\\"quarterly\\"><label for=\\"quarterly\\">Quarterly</label></div><div><input type=\\"radio\\" id=\\"semiannually\\" name=\\"donation-frequency\\" value=\\"semiannually\\"><label for=\\"semiannually\\">Semiannually</label></div><div><input type=\\"radio\\" id=\\"annually\\" name=\\"donation-frequency\\" value=\\"annually\\"><label for=\\"annually\\">Annually</label></div></div><p class=\\"donate-text-button\\">Donate $ XXXX to Charity 1 every XX month(s)</p><div><button class=\\"donate-page-button custom-shadow-button\\">Donate with Debit or Credit Card</button></div></div></div>"`;
